plugins/services/src/js/utils/GraphQLTaskUtil.js

```diff
@@ -96,15 +99,42 @@ function mergeHostname(task) {
   return task;
 }
 
+function mergeData(task, flattenTaskTree) {
+  // Merge version from Marathon
+  task = mergeVersion(task, flattenTaskTree);
+  // Get Health from Mesos first, and fallback on Marathon
+  task = mergeHealth(task, flattenTaskTree);
+  // Merge hostname if we can find it
+  task = mergeHostname(task);
+
+  return task;
+}
+
+function buildFlatServiceTree(serviceTree) {
```

Comment:
I would check if I find the place for it, otherwise, we should discuss a little bit where it would be the best place to trigger it. My concern is that before mergeData I am sure I have the most recent version of the tree.

Reply:
I moved all the code to the Service now. So I am trying to figure this out.